pydantic/schema.py

```diff
@@ -0,0 +1,392 @@
+from datetime import date, datetime, time, timedelta
+from decimal import Decimal
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, Sequence, Set, Tuple, Type
+from uuid import UUID
+
+from . import main
+from .fields import Field, Shape
+from .json import pydantic_encoder
+from .types import DSN, UUID1, UUID3, UUID4, UUID5, DirectoryPath, EmailStr, FilePath, Json, NameEmail, UrlStr
+from .utils import clean_docstring
+
+
+def get_flat_models_from_model(model: Type['main.BaseModel']) -> Set[Type['main.BaseModel']]:
+    flat_models: Set[Type[main.BaseModel]] = set()
+    assert issubclass(model, main.BaseModel)
+    flat_models.add(model)
+    for field in model.__fields__.values():
+        flat_models = flat_models | get_flat_models_from_field(field)
```

Comment:
`flat_models |= get_flat_models_from_field(field)` and below

Reply:
Done